### PR TITLE
Enhanced RDoc for String#succ

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
@jeremyevans, the penultimate collection of examples are not what I expected.  Rollover seems not to pass through  255.chr as I'd expected.

File: string.c [hunk 1]
```diff
@@ -4160,27 +4160,59 @@ static VALUE str_succ(VALUE str);
...
- *  adding an additional character if necessary.
- *
- *     "abcd".succ        #=> "abce"
- *     "THX1138".succ     #=> "THX1139"
- *     "<<koala>>".succ   #=> "<<koalb>>"
- *     "1999zzz".succ     #=> "2000aaa"
- *     "ZZZ9999".succ     #=> "AAAA0000"
- *     "***".succ         #=> "**+"
+ *    string.succ -> new_str
+ *    string.next -> new_str
```
Comment (jeremyevans) [inline string.c:4164]: I think we should leave `next` off the call-seq, and only mention it near the end as an alias.

**Review by jeremyevans — APPROVED** [review]
Looks good.  I think we should adjust the call-seq to not mention the alias, but it's up to you.

File: string.c (hunk 1 above)
Comment (BurdetteLamar) [reply string.c:4164]: @jeremyevans, the penultimate collection of examples are not what I expected. Rollover seems not to pass through 255.chr as I'd expected.  So the text needs to change.  I'll draft something.

**Review by jeremyevans — APPROVED** [review]
Looks good, I like the replaced examples.

**Pull request merged**